### PR TITLE
feat(git): Preserve branch creation time as git author date

### PR DESCRIPTION
**Pull request by zharinov**
Ref: #7387

### CODE REVIEW

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -50,6 +50,11 @@ interface LocalConfig extends StorageConfig {
   branchPrefix: string;
 }
 
+enum GitDate {
+  Author = '%ai',
```
Comment (zharinov) [inline lib/util/git/index.ts:54]: Now it's constant part

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -50,6 +50,11 @@ interface LocalConfig extends StorageConfig {
...
+  Committer = '%ci',
```
Comment (zharinov) [inline lib/util/git/index.ts:55]: And this one is moving part

File: lib/util/git/index.ts [hunk 3]
```diff
@@ -604,9 +622,19 @@ export async function commitFiles({
...
+    );
+    const committerDate = new Date();
+    const commitRes = await git
+      .env({
+        GIT_COMMITTER_DATE: committerDate.toISOString(),
+      })
+      .commit(message, [], {
+        '--no-verify': null,
+        '--date': authorDate.toISOString(),
+      });
```
Comment (zharinov) [inline lib/util/git/index.ts:637]: We're setting both dates explicitly, so that date format is same

**Comment by zharinov** [comment]
@rarkins @viceice @JamieMagee 
What do you think can go wrong? :)

**Comment by zharinov** [comment]
Need to handle branch creation date

**Comment by rarkins** [comment]
Can you explain what it's doing?

**Comment by zharinov** [comment]
The idea is to set author date at branch creation and them maintain it with the next (force) commits. If we need to know actual commit date, we use commiter date instead. But author date always stays same, so we can query and use it to limit branches created per hour.

**Comment by zharinov** [comment]
Logically, committer date should go after author's one, so I guess it can be used to store actual commit time when we need it.

**Comment by zharinov** [comment]
Here is how it looks for [my test PR](https://github.com/zharinov/renovate-test-hourly/pull/3):
```
commit 35779c55f9edd9797ac4e2edede4564c8167733d
Author:     Renovate Bot <renovate@whitesourcesoftware.com>
AuthorDate: Fri Oct 2 13:06:18 2020 +0000
Commit:     Renovate Bot <renovate@whitesourcesoftware.com>
CommitDate: Fri Oct 2 14:04:07 2020 +0000

```

**Comment by rarkins** [comment]
I'm not sure this is a good idea because we could end up with scenarios where the author date shows as hours or days earlier than a version was even released. e.g.

monday: PR to update to v1.0.2 is created
tuesday: v1.0.3 is released, Renovate force pushes but keeps monday's author date

Should we change our focus to instead be commits per repo per hour? After all, anyone using this limit likely is doing so in order to not trigger too many CI pipelines per hour, and commits will typically do that regardless of whether it's a new branch or a force push to an existing.

**Comment by zharinov** [comment]
Another bad case I'm not happy about if branch was created, merged and deleted in same hour, which means we can't count it.

**Pull request closed without merging**